lib/ansible/modules/cloud/vmware/vmware_host_auto_start.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, sky-joker
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+module: vmware_host_auto_start
+short_description: Manage the auto power ON or OFF for vm on ESXi host
+author:
+  - sky-joker (@sky-joker)
+version_added: '2.10'
+description:
+  - In this module, can set up automatic startup and shutdown of virtual machines according to host startup or shutdown.
+requirements:
+  - python >= 2.7
+  - PyVmomi
+options:
+  esxi_hostname:
+    description:
+    - ESXi hostname where the VM to set auto power on or off exists.
+    type: str
+    required: True
+  name:
```

Comment:
Could you please add `moid` and `use_instance_uuid` parameters to match other modules which finds the VM in vCenter environment?

Reply:
I added `moid` and `use_instance_uuid` parameters.